doc/pages/commands.asciidoc

```diff
@@ -299,6 +299,15 @@ but not really useful in that context.
 *debug* {info,buffers,options,memory,shared-strings,profile-hash-maps,faces,mappings}::
     print some debug information in the *\*debug** buffer
 
+== Module commands
+
+*provide-module* [<switches>] <name> <commands>::
+    declares a new module that is provided by the given commands. The commands
+    passed into the module are guaranteed to be run only once.
```

Comment:
This would be the primary documentation for this feature, so it's worth going into a bit of detail. How about:

> Declares that the module <name> is defined by <commands>. <commands> will be evaluated as if by `evaluate-commands` the first time `require-module <name>` is run.

Actually, *is* it evaluated as if by `evaluate-commands`? Does it execute in a draft context? Does it execute in a particular client context? Does it have access to buffer or window scopes, or does that depend on where the `require-module` command is executed? If so, should it be restricted to global scope, so that we don't get weird bugs where behaviour depends on which `require-module` statement executes first?

Reply:
Also, this command fails if the same module is defined twice, yes? That should be mentioned too.